app/models/filterable_search.rb

```diff
@@ -42,19 +44,35 @@ def default_sort?
 
   def initialize_date_attributes(options)
     @until = parse_until_ts(options[:until_date])
-    @since = parse_since_ts(options[:since_date])
+    @since = parse_since_ts(options[:since_date], @until)
+    @created_until = parse_until_ts(options[:created_until_date])
+    @created_since = parse_since_ts(options[:created_since_date], @created_until)
 
-    if @since && @until && @since > @until
-      @since, @until = @until.beginning_of_day, @since.end_of_day
-    end
+    flip_reversed_dates(@since, @until, @created_since, @created_until)
+
+    return unless (@since.nil? && @until.nil?) || (@created_since.nil? && @created_until.nil?)
 
+    time_based_search(options)
```

Comment:
Since `#time_based_search` only needs `options[:tbs]`, we could pass only that option into the method instead of the full `options`.

Reply:
Good call.